lib/plugins/jsonBodyParser.js

```diff
@@ -20,40 +20,48 @@ var bodyReader = require('./bodyReader');
  */
 function jsonBodyParser(options) {
     assert.optionalObject(options, 'options');
-    options = options || {};
+    var opts = options || {};
 
-    var override = options.overrideParams;
+    var override = opts.overrideParams;
 
     function parseJson(req, res, next) {
         if (req.getContentType() !== 'application/json' || !req.body) {
-            next();
-            return;
+            return next();
         }
 
         var params;
 
         try {
-            params = JSON.parse(req.body, options.reviver);
+            params = JSON.parse(req.body, opts.reviver);
         } catch (e) {
-            next(new errors.InvalidContentError('Invalid JSON: ' +
+            return next(new errors.InvalidContentError('Invalid JSON: ' +
                 e.message));
-            return;
         }
 
-        if (options.mapParams !== false) {
+        if (opts.mapParams === true) {
             if (Array.isArray(params)) {
+                // if req.params exists, we have url params. we can't map an
+                // array safely onto req.params, throw an error.
+                if (req.params &&
```

Comment:
Yeah - req.params defaults to false now, so this logic handles some of the weird use cases and fixes existing issues where mapParams stomps over url variables.